rpcclient/infrastructure.go

```diff
@@ -809,6 +809,16 @@ func (c *Client) handleSendPostMessage(jReq *jsonRequest) {
 		jReq.responseChan <- &Response{err: err}
 		return
 	}
+	// We still want to return an error if for any reason the respone
+	// remains empty.
+	if httpResponse == nil {
+		jReq.responseChan <- &Response{
+			err: fmt.Errorf("invalid http POST response (nil), "+
+				"method: %s, id: %d, last error=%v",
+				jReq.method, jReq.id, lastErr),
```

Comment:
Looks like this diff did not cherry-pick cleanly.  Some prerequisite commits were not included: 
https://github.com/btcsuite/btcd/pull/1856 and https://github.com/btcsuite/btcd/pull/1862 both preceeded this